etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+*
+* @name no-sparse-arrays
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-sparse-arrays]{@link https://eslint.org/docs/rules/no-sparse-arrays}
+*/
+rules[ 'no-sparse-arrays' ] = 'error';
+
+/**
+* Disallows template literal placeholder syntax in regular strings.
+*
+* @name no-template-curly-in-string
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-template-curly-in-string]{@link https://eslint.org/docs/rules/no-template-curly-in-string}
+*/
+rules[ 'no-template-curly-in-string' ] = 'error';
+
+/**
+* Disallows throwing literals as exceptions.
+*
+* @name no-throw-literal
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-throw-literal]{@link https://eslint.org/docs/rules/no-throw-literal}
+*/
+rules[ 'no-throw-literal' ] = 'error';
+
+/**
+* Disallows trailing whitespace at the end of lines.
+*
+* @name no-trailing-spaces
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-trailing-spaces]{@link https://eslint.org/docs/rules/no-trailing-spaces}
+*/
+rules[ 'no-trailing-spaces' ] = 'error';
+
+/**
+* Disallows initializing variables to `undefined`.
+*
+* @name no-undef-init
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-undef-init]{@link https://eslint.org/docs/rules/no-undef-init}
+*/
+rules[ 'no-undef-init' ] = 'error';
+
+/**
+* Disallows dangling underscores in identifiers.
+*
+* @name no-underscore-dangle
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-underscore-dangle]{@link https://eslint.org/docs/rules/no-underscore-dangle}
+*/
+rules[ 'no-underscore-dangle' ] = 'error';
+
+/**
+* Disallows control flow statements in `finally` blocks.
+*
+* @name no-unsafe-finally
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unsafe-finally]{@link https://eslint.org/docs/rules/no-unsafe-finally}
+*/
+rules[ 'no-unsafe-finally' ] = 'error';
+
+/**
+* Disables the rule against unused expressions.
+*
+* @name no-unused-expressions
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unused-expressions]{@link https://eslint.org/docs/rules/no-unused-expressions}
+*/
+rules[ 'no-unused-expressions' ] = 'off';
+
+/**
+* Disallows unused labels.
+*
+* @name no-unused-labels
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unused-labels]{@link https://eslint.org/docs/rules/no-unused-labels}
+*/
+rules[ 'no-unused-labels' ] = 'error';
+
+/**
+* Disables the rule against unused variables.
+*
+* @name no-unused-vars
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unused-vars]{@link https://eslint.org/docs/rules/no-unused-vars}
+*/
+rules[ 'no-unused-vars' ] = 'off';
+
+/**
+* Disables the rule against using variables before they are defined.
+*
+* @name no-use-before-define
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-use-before-define]{@link https://eslint.org/docs/rules/no-use-before-define}
+*/
+rules[ 'no-use-before-define' ] = 'off';
```

Comment:
Is this to avoid conflict with a prior rule which bars this behavior?

Reply:
We enable `@typescript-eslint/no-use-before-define` instead. But I removed this section. Not necessary to turn it off explicitly again.